src/porepy/models/fluid_mass_balance.py

```diff
@@ -392,63 +402,20 @@ def bc_type_mobrho(self, sd: pp.Grid) -> pp.BoundaryCondition:
         # Define boundary condition on all boundary faces.
         return pp.BoundaryCondition(sd, boundary_faces, "dir")
 
-    def bc_values_darcy(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:
-        """Boundary condition values for the Darcy flux.
-
-        Parameters:
-            subdomains: List of subdomains.
-
-        Returns:
-            Ad array representing the boundary condition values for the Darcy flux.
-
-        """
-        num_faces = sum([sd.num_faces for sd in subdomains])
-        # Ignore typing error below, the parameter in _ad_wrapper forces it to be an
-        # Array.
-        return pp.wrap_as_ad_array(0, num_faces, "bc_values_darcy")
-
-    def bc_values_mobrho(self, subdomains: list[pp.Grid]) -> pp.ad.DenseArray:
```

Comment:
In my understanding, Neumann-type conditions for conservation equations can either be specified as the total influx of the conserved quantity (so and so much mass enters through this face), or by specifying the influx of fluid together with the concentration of the conserved quantitiy (this would be a Neumann condition for the Darcy flux, but a Dirichlet-type condition for the density or similar). I think we need to be able to do both. What that means in terms of methods and return types, I am not sure of.

Reply:
I guess, now it works as it should. There are 3 cases for the single phase fluid flow (the logic is the same for the thermal model):
1. Neumann `bc_type_mobrho` - we're setting the total influx of the conserved quantity.
2. Dirichlet `bc_type_mobrho`, Neumann `bc_type_darcy` - we're setting the fluid volumetric influx and pressure. It is multiplied by `mobility_rho` which takes pressure values.
3. Dirichlet `bc_type_mobrho` and `bc_type_darcy` - we're setting pressure values.

Reply:
I think we can resolve this for now, and then hope that the bigger question of consistency between boundary conditions will be addressed in the near future. @IvarStefansson Do you agree?